netlify/functions/utils/tmdb.ts

```diff
@@ -2,66 +2,100 @@ import axios from "axios";
+  return axios.get(`https://api.themoviedb.org/3${path}?api_key=${tmdbApiKey}`);
 }
 
 async function getTMDBConfig() {
-    return makeTMDBApiCall("/configuration");
+  return makeTMDBApiCall("/configuration");
 }
 
 export async function getTMDBMovieData(movieId: number) {
-    return makeTMDBApiCall(`/movie/${movieId}`);
+  return makeTMDBApiCall(`/movie/${movieId}`);
 }
 
 export async function getMovieData(movieId: number) {
-    const configuration = await getTMDBConfig()
-    const movieData = await getTMDBMovieData(movieId)
+  const configuration = await getTMDBConfig();
+  const movieData = await getTMDBMovieData(movieId);
 
-    movieData.data.poster_url = configuration.data.images.base_url + "w500" + movieData.data.poster_path
+  movieData.data.poster_url =
+    configuration.data.images.base_url + "w500" + movieData.data.poster_path;
 
-    return movieData.data
+  return movieData.data;
 }
 
-export async function getReviewData(reviews: ReviewDatabaseObject[]): Promise<ReviewResponse[]> {
-    return Promise.all(
-        reviews.map(async (review) => {
-            const response = await makeTMDBApiCall(`/movie/${review.movieId}`);
-            return {
-                ...review,
-                movieTitle: response.data.title,
-            };
-        })
-    );
+export async function getReviewData(
+  reviews: ReviewDatabaseObject[]
+): Promise<ReviewResponse[]> {
+  return Promise.all(
+    reviews.map(async (review) => {
+      const response = await makeTMDBApiCall(`/movie/${review.movieId}`);
+      return {
+        ...review,
+        movieTitle: response.data.title,
+      };
+    })
+  );
 }
 
-export async function getDetailedReviewData(reviews: DetailedReviewResponse[]): Promise<DetailedReviewResponse[]> {
-    const configuration = await getTMDBConfig();
+export async function getDetailedReviewData(
+  reviews: DetailedReviewResponse[]
+): Promise<DetailedReviewResponse[]> {
+  const configuration = await getTMDBConfig();
 
-    await Promise.all(
-        reviews.map(async (review) => {
-            const response = await makeTMDBApiCall(`/movie/${review.movieId}`);
-            review.movieTitle = response.data.title;
-            review.movieData = response.data;
-            review.movieData.poster_url = `${configuration.data.images.secure_base_url}w154${response.data.poster_path}`;
-        })
-    );
-    return reviews;
+  await Promise.all(
+    reviews.map(async (review) => {
+      const response = await makeTMDBApiCall(`/movie/${review.movieId}`);
+      review.movieTitle = response.data.title;
+      review.movieData = response.data;
+      review.movieData.poster_url = `${configuration.data.images.secure_base_url}w154${response.data.poster_path}`;
+    })
+  );
+  return reviews;
 }
 
-export async function getWatchlistItemMovieData(watchList: WatchListItem[]): Promise<WatchListItem[]> {
-    const configuration = await getTMDBConfig();
+export async function getDetailedMovie<T extends BaseMovie>(
```

Comment:
It seems like there is potential to share code between `getDetailedMovie` and `getDetailedReviewData`

Reply:
Yes, I wrote this so it could be used in place of `getDetailedReviewData` out of the box, I just forgot to swap it out in the reviews endpoint. Updated. Now the reviews endpoint also uses this function.